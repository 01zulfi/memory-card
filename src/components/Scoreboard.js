import React from "react";

const Scoreboard = ({ currentScore, bestScore }) => {
  return (
    <div className="scoreboard">
      <div className="current-score-div">
        <p>Current Score</p>
        <p>{currentScore}</p>
      </div>
      <div className="best-score-div">
        <p>Best Score</p>
        <p>{bestScore}</p>
      </div>
    </div>
  );
};

export default Scoreboard;
