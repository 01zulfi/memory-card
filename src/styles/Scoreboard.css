div.scoreboard {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  width: 100%;
  margin-top: 10px;
  background: rgb(171, 237, 198, 0.7);
}

div.scoreboard > div + div {
  display: flex;
  flex-direction: column;
}

p {
  text-align: center;
}

p.score-heading {
  color: #1b2021;
  font-size: 19px;
}

p.score-display {
  font-weight: bolder;
  font-size: 25px;
}

@media (min-width: 768px) {
  p.score-display {
    font-size: 30px;
  }
}
