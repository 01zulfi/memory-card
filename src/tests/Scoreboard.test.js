import React from "react";
import Scoreboard from "../components/Scoreboard";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";

test("scoreboard renders", () => {
  const { container } = render(<Scoreboard />);
  expect(container).toMatchSnapshot();
  expect(container.firstChild).toHaveClass("scoreboard");
});

test("scoreboard renders correct values", () => {
  const { getByText } = render(<Scoreboard currentScore={2} bestScore={5} />);
  const currentScore = getByText("2");
  const bestScore = getByText("5");

  expect(currentScore).toBeInTheDocument();
  expect(currentScore.parentNode).toHaveClass("current-score-div");
  expect(bestScore).toBeInTheDocument();
  expect(bestScore.parentNode).toHaveClass("best-score-div");
});
