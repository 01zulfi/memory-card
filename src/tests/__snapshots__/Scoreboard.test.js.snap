// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scoreboard renders 1`] = `
<div>
  <div
    class="scoreboard"
  >
    <div
      class="current-score-div"
    >
      <p>
        Current Score
      </p>
      <p />
    </div>
    <div
      class="best-score-div"
    >
      <p>
        Best Score
      </p>
      <p />
    </div>
  </div>
</div>
`;
